### PR TITLE
DEV Refactor processing function in each class

### PR DESCRIPTION
**Pull request by thomasyu888**
* Correct kwargs usage
* Use of read_file
* Make more pythonic (Update imports)

### CODE REVIEW

File: genie/maf.py [hunk 1]
```diff
@@ -50,72 +51,82 @@ def storeProcessedMaf(self, filePath, mafSynId, centerMafSynId, isNarrow=False):
...
-            if self._fileType == "maf":
-                self.storeProcessedMaf(narrowMafPath, mafSynId, centerMafSynId, isNarrow=True)
-            #Store MAF flat file into synapse
-            self.storeProcessedMaf(newMafPath, mafSynId, centerMafSynId)
-        else:
-            logger.error('ERROR PROCESSING %s' % filePath)
-            filePath = "NOTPROCESSED"
-        return(filePath)
+    def preprocess(self, filePath, **kwargs):
+        databaseToSynIdMappingDf = kwargs['databaseToSynIdMappingDf']
```
Comment (zimingd) [inline genie/maf.py]: No need to preprocess

File: genie/maf.py [hunk 2]
```diff
@@ -50,72 +51,82 @@ def storeProcessedMaf(self, filePath, mafSynId, centerMafSynId, isNarrow=False):
...
+                           "--output-maf",newMafPath,
+                           "--vep-fork", '8',
+                           "--tmp-dir",tempdir,
+                           '--vep-path', veppath,
+                           '--vep-data', vepdata,
+                           #'--ref-fasta','/root/.vep/homo_sapiens/86_GRCh37/Homo_sapiens.GRCh37.75.dna.primary_assembly.fa',
+                           "--custom-enst", os.path.join(vcf2mafPath,"data/isoform_overrides_uniprot")]
+            if reference is not None:
+                commandCall.extend(["--ref-fasta",reference])
+            maf = subprocess.call(commandCall)
```
Comment (zimingd) [inline genie/maf.py]: Use check call

File: genie/maf.py [hunk 3]
```diff
@@ -50,72 +51,82 @@ def storeProcessedMaf(self, filePath, mafSynId, centerMafSynId, isNarrow=False):
...
+                logger.error('ERROR PROCESSING %s' % filePath)
+                filePath = "NOTPROCESSED"
         else:
             logger.info("Please run with `--process %s` parameter if you want to reannotate the %s files" % (self._fileType, self._fileType))
-        return(mutationFiles)
+        return(filePath)
+
+    # def process_steps(self, filePath, **kwargs): 
+    #   processing = kwargs['processing']
+    #   mutationFiles = []
```
Comment (zimingd) [inline genie/maf.py]: Remove

File: genie/clinical.py [hunk 4]
```diff
@@ -158,10 +160,7 @@ def preprocess(self, filePath, **kwargs):
         databaseToSynIdMappingDf = kwargs['databaseToSynIdMappingDf']
```
Comment (thomasyu888) [inline genie/clinical.py]: No need for preprocess

**Comment by thomasyu888** [comment]
Pull in #75  first

**Review by zimingd — APPROVED** [review]

**Pull request merged**